test-clients/src/test/java/E2EPackageRunner.java

```diff
@@ -0,0 +1,791 @@
+import com.hedera.services.bdd.suites.token.TokenMiscOps;
+import com.hedera.services.bdd.suites.token.TokenPauseSpecs;
+import com.hedera.services.bdd.suites.token.TokenTotalSupplyAfterMintBurnWipeSuite;
+import com.hedera.services.bdd.suites.token.TokenTransactSpecs;
+import com.hedera.services.bdd.suites.token.TokenUpdateSpecs;
+import com.hedera.services.bdd.suites.token.UniqueTokenManagementSpecs;
+import com.hedera.services.bdd.suites.utils.FetchWithPem;
+import com.hedera.services.bdd.suites.utils.sysfiles.SysFilesUpdate;
+import com.hedera.services.bdd.suites.utils.validation.OneOffValidation;
+import com.hedera.services.bdd.suites.utils.validation.ValidationScenarios;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DynamicContainer;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.TestFactory;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Supplier;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.DynamicContainer.dynamicContainer;
+import static org.junit.jupiter.api.DynamicTest.dynamicTest;
+
+
+class E2EPackageRunner {
+
+	@BeforeAll
+	static void beforeAll() {
+		final var portSystemProperty = System.getProperty("DEFAULT_PORT");
+		final var defaultPort = portSystemProperty != null ? portSystemProperty : "50211";
+		final var defaultProperties = JutilPropertySource.getDefaultInstance();
+		HapiApiSpec.runInCiMode(
+				defaultPort + ":50212",
+				defaultProperties.get("default.payer"),
+				defaultProperties.get("default.node").split("\\.")[2],
+				defaultProperties.get("tls"),
+				defaultProperties.get("txn.proto.structure"),
+				defaultProperties.get("node.selector"),
+				Collections.emptyMap()
+		);
+	}
+
+	@Tag("autorenew")
+	@TestFactory
+	Collection<DynamicContainer> autorenew() {
+		return List.of(
+				extractSpecsFromSuite(AccountAutoRenewalSuite::new),
+				extractSpecsFromSuite(AutoRemovalCasesSuite::new),
+				extractSpecsFromSuite(GracePeriodRestrictionsSuite::new),
+				extractSpecsFromSuite(MacroFeesChargedSanityCheckSuite::new),
+				extractSpecsFromSuite(NoGprIfNoAutoRenewSuite::new),
+				extractSpecsFromSuite(TopicAutoRenewalSuite::new)
+		);
+	}
+
+	@Tag("compose")
+	@TestFactory
+	Collection<DynamicContainer> compose() {
+		return List.of(
+				extractSpecsFromSuite(LocalNetworkCheck::new),
+				extractSpecsFromSuite(PerpetualLocalCalls::new)
+		);
+	}
+
+	@Tag("consensus")
+	@TestFactory
+	Collection<DynamicContainer> consensus() {
+		return List.of(
+				extractSpecsFromSuite(AssortedHcsOps::new),
+				extractSpecsFromSuite(ChunkingSuite::new),
+				extractSpecsFromSuite(SubmitMessageSuite::new),
+				extractSpecsFromSuite(TopicCreateSuite::new),
+				extractSpecsFromSuite(TopicDeleteSuite::new),
+				extractSpecsFromSuite(TopicGetInfoSuite::new),
+				extractSpecsFromSuite(TopicUpdateSuite::new)
+		);
+	}
+
+	@Tag("contract.precompile")
```

Comment:
Might be useful to add multiple tags in case we want to run all the contract tests, crypto tests, etc.
```suggestion
	@Tag("contract")
	@Tag("contract.precompile")
```